dbt/.gitignore

```diff
@@ -5,3 +5,4 @@ target/
 dbt_modules/
 dbt_packages/
 assets/*.svg
+master-cache/
```

Comment:
The docs now recommend storing the main branch state in this subdirectory, so we add it to `.gitignore` to ensure that it does not get committed accidentally.